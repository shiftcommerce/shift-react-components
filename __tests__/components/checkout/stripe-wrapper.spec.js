// Libraries
import React from 'react'
import { StripeProvider } from 'react-stripe-elements'

// Lib
import Config from '../../../src/lib/config'

// Component
import StripeWrapper from '../../../src/components/checkout/stripe-wrapper'

describe('StripeWrapper', () => {
  it('renders the payment method is not available if api key is not present', () => {
    // Act
    const wrapper = shallow(<StripeWrapper />)

    // Assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper).toIncludeText('* PAYMENT THROUGH STRIPE IS CURRENTLY NOT AVAILABLE *')
  })

  it('renders StripeProvider component if the api key is present', () => {
    // Arrange
    Config.set({
      stripeApiKey: 'TEST_API_KEY'
    })

    // Act
    const wrapper = shallow(<StripeWrapper />)

    // Assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(StripeProvider).length).toEqual(1)
  })
})
