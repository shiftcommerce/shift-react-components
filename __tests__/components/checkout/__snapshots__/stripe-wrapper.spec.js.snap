// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StripeWrapper renders StripeProvider component if the api key is present 1`] = `
<Provider
  apiKey="TEST_API_KEY"
>
  <Elements>
    <InjectStripe(StripeFields) />
  </Elements>
</Provider>
`;

exports[`StripeWrapper renders the payment method is not available if api key is not present 1`] = `
<div
  className="u-bold u-text-color--red"
>
  * PAYMENT THROUGH STRIPE IS CURRENTLY NOT AVAILABLE *
</div>
`;
